src/Traits/DetectsChanges.php

```diff
@@ -150,6 +150,10 @@ protected static function getRelatedModelAttributeValue(Model $model, string $at
 
         [$relatedModelName, $relatedAttribute] = explode('.', $attribute);
 
+        if (Str::contains($relatedModelName, '_')) {
```

Comment:
I haven't tested but couldn't we run the camel transformation everytime? Or does it destroy already camelized strings? 🤔

Reply:
You're right, we can just camelize every time 😄